src/gt4py/next/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -497,6 +516,34 @@ def visit_SetAt(
         stencil = node.expr.fun.args[0]  # type: ignore[attr-defined] # checked in assert
         domain = node.domain
         inputs = node.expr.args
+        lowered_inputs = []
+        for input_ in inputs:
+            lowered_input = self.visit(input_, **kwargs)
+
+            def _convert_input(
```

Comment:
- unclear why this should be a closure
- `_convert_input` -> better name

Reply:
Refactored this section and the `convert_output` case below to use a common utility function `_process_elements`.